src/error.rs

```diff
@@ -59,12 +63,18 @@ impl Error {
 
     /// If other.pointer is NULL, free self ; otherwise, moves self into other. The error variable
     /// other.pointer points to must be NULL.
-    /// 
+    ///
     /// Note that self is no longer valid after this call. If you want to keep using the same Error
     /// struct, you need to set it to NULL after calling this function on it.
     pub fn propagate(&mut self, other: &Error) -> () {
         unsafe { ffi::g_propagate_error(&mut self.pointer, other.pointer) }
     }
+
+    /// Returns the error message stored in the wrapped GError
+    pub fn message(&self) -> &str {
+        let c_str = unsafe { CStr::from_ptr((*self.pointer).message) };
```

Comment:
We can't guarantee that the `message` lives as long as `&self`. For example, the `set` method will invalidate the pointer.